Sources/web3swift/Web3/Web3+Structures.swift

```diff
@@ -689,3 +689,58 @@ public struct TxPoolContentForNonce {
     public var nonce: BigUInt
     public var details: [TransactionDetails]
 }
+
+public enum FilterChanges: Decodable {
+    case hashes([String])
+    case logs([EventLog])
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        if let hashes = try? container.decode([String].self) {
+            self = .hashes(hashes)
+        } else {
+            self = .logs(try container.decode([EventLog].self))
+        }
+    }
+}
+
+public struct BlockHeader: Decodable {
+    public let difficulty: String
+    public let extraData: String
+    public let gasLimit: String
+    public let gasUsed: String
+    public let logsBloom: String
+    public let miner: String
+    public let nonce: String
+    public let number: String
+    public let parentHash: String
+    public let receiptsRoot: String
+    public let sha3Uncles: String
+    public let stateRoot: String
+    public let timestamp: String
+    public let transactionsRoot: String
+}
```

Comment:
As an alternative I suggest modifying this function to accept a generic type: https://github.com/skywinder/web3swift/pull/446/files#diff-9743ac53f1b42c7462dbf4d5dde6f2f434845bf1bdc66c25858bf6b4cdacf0f9R27

Depending on the network used model can be specified and it will be up to the user of this library.

Reply:
I'm also researching this part of the lib right now and i've confused either.

In web3js regarding to the docs these 4 fields are optional, in the ether.js all fields are non optional.

More than that in Ethereum github JSON-RPC schema all of the fields are marked as required.

Speaking about go implementation, as i saw their code it seems that explicitly optional checking for external API struct fields are best practices for that language, which is not in swift.